src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -426,75 +394,36 @@ CitusModifyBeginScan(CustomScanState *node, EState *estate, int eflags)
 	if (workerJob->requiresMasterEvaluation)
 	{
 		PlanState *planState = &(scanState->customScanState.ss.ps);
+		EState *executorState = planState->state;
 
 		ExecuteMasterEvaluableFunctions(jobQuery, planState);
 
+		/* we've already processed parameters */
+		executorState->es_param_list_info = NULL;
```

Comment:
@marcocitus if you have a quick answer, let me know but I think the following applies…

I believe that if we don't set this to `NULL`, then within either of the `Execute*ModifyTask(s)` functions, we'd find parameter info and serialize and send them over the wire. I'm not sure whether this actually causes any bad behavior, but it will definitely be wasted effort if we've already evaluated parameters and baked them into the SQL.

However, this is just my impression from reading. I think the comment deserves some explanation.